docs/subset09.md

```diff
@@ -89,7 +89,11 @@ union-type-desc =
   optional-type-desc
   | union-type-desc "|" optional-type-desc
 
-optional-type-desc = builtin-type-name [ "?" ]
+optional-type-desc = builtin-type-desc [ "?" ]
```

Comment:
Earlier I wrote it as `optional-type-desc = (nil-type-desc | builtin-type-name) [ "?" ]` but when I am implementing this realized that I do need a name for `nil-type-desc | builtin-type-name` and might as well write the grammar the same way.

@jclark is this a good name?